File: src/test/java/no/nav/sbl/MainTest.java
package no.nav.sbl;

import no.nav.common.nais.NaisYamlUtils;
import no.nav.common.rest.client.RestClient;
import no.nav.common.test.SystemProperties;
import no.nav.common.test.ssl.SSLTestUtils;
import no.nav.common.test.ssl.TrustAllSSLSocketFactory;
import org.springframework.boot.SpringApplication;

import javax.net.ssl.X509TrustManager;
import java.io.IOException;
import java.security.cert.CertificateException;
import java.security.cert.X509Certificate;
import java.util.HashMap;

public class MainTest {
    static {
        setupRestClient();
        SystemProperties.setFrom(".vault.properties");
        NaisYamlUtils.NaiseratorSpec naisConfig = NaisYamlUtils.getTemplatedConfig(".nais/qa-template.yaml", new HashMap<>() {{
            put("namespace", "q0");
        }});
        NaisYamlUtils.loadFromYaml(naisConfig);
        SSLTestUtils.disableCertificateChecks();
    }

    public static void main(String[] args) throws IOException {
        SpringApplication application = new SpringApplication(Main.class);
        application.setAdditionalProfiles("local");
        application.run(args);
    }

    private static void setupRestClient() {
        RestClient.setBaseClient(RestClient.baseClientBuilder()
                .sslSocketFactory(new TrustAllSSLSocketFactory(), new X509TrustManager() {
                    public void checkClientTrusted(X509Certificate[] x509Certificates, String s) throws CertificateException {
                    }

                    public void checkServerTrusted(X509Certificate[] x509Certificates, String s) throws CertificateException {
                    }

                    public X509Certificate[] getAcceptedIssuers() {
                        return new X509Certificate[0];
                    }
                })
                .build());
    }

}

File: src/test/java/no/nav/sbl/MainTest.kt
package no.nav.sbl

import no.nav.common.nais.NaisYamlUtils
import no.nav.common.rest.client.RestClient
import no.nav.common.test.ssl.SSLTestUtils
import no.nav.common.test.ssl.TrustAllSSLSocketFactory
import org.springframework.boot.SpringApplication
import java.io.IOException
import java.security.cert.CertificateException
import java.security.cert.X509Certificate
import javax.net.ssl.X509TrustManager

object MainTest {
    init {
        setupRestClient()
        val naisConfig =
            NaisYamlUtils.getTemplatedConfig(
                ".nais/dev.yaml",
                object : HashMap<Any?, Any?>() {
                    init {
                        put("namespace", "q0")
                    }
                },
            )
        NaisYamlUtils.loadFromYaml(naisConfig)
        SSLTestUtils.disableCertificateChecks()
    }

    @Throws(IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val application = SpringApplication(Main::class.java)
        application.setAdditionalProfiles("local")
        application.run(*args)
    }

    private fun setupRestClient() {
        RestClient.setBaseClient(
            RestClient
                .baseClientBuilder()
                .sslSocketFactory(
                    TrustAllSSLSocketFactory(),
                    object : X509TrustManager {
                        @Throws(CertificateException::class)
                        override fun checkClientTrusted(
                            x509Certificates: Array<X509Certificate>,
                            s: String,
                        ) {
                        }

                        @Throws(CertificateException::class)
                        override fun checkServerTrusted(
                            x509Certificates: Array<X509Certificate>,
                            s: String,
                        ) {
                        }

                        override fun getAcceptedIssuers(): Array<X509Certificate> = emptyArray()
                    },
                ).build(),
        )
    }
}
